v24/src/main/java/com/foursoft/harness/kbl/v24/HasConnectionOrOccurrences.java

```diff
@@ -0,0 +1,106 @@
+
+    List<KblConnectorOccurrence> getConnectorOccurrences();
+
+    List<KblFixingOccurrence> getFixingOccurrences();
+
+    List<KblGeneralWireOccurrence> getGeneralWireOccurrences();
+
+    List<KblSpecialTerminalOccurrence> getSpecialTerminalOccurrences();
+
+    List<KblTerminalOccurrence> getTerminalOccurrences();
+
+    List<KblWireProtectionOccurrence> getWireProtectionOccurrences();
+
+    List<KblWiringGroup> getWiringGroups();
+
+    default List<ConnectionOrOccurrence> getConnectionOrOccurrences() {
+        return Stream.of(getAccessoryOccurrences(),
+                        getAssemblyPartOccurrences(),
+                        getCavityPlugOccurrences(),
+                        getCavitySealOccurrences(),
+                        getCoPackOccurrences(),
+                        getComponentBoxOccurrences(),
+                        getComponentOccurrences(),
+                        getConnections(),
+                        getConnectorOccurrences(),
+                        getFixingOccurrences(),
+                        getGeneralWireOccurrences(),
+                        getSpecialTerminalOccurrences(),
+                        getTerminalOccurrences(),
+                        getWireProtectionOccurrences(),
+                        getWiringGroups())
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList());
+    }
+
+    default List<HasRelatedAssembly> getHasRelatedAssemblyOccurrences() {
+        return Stream.of(getAccessoryOccurrences().stream().flatMap(StreamUtils.ofClass(HasRelatedAssembly.class)),
+                        getCavityPlugOccurrences().stream().flatMap(StreamUtils.ofClass(HasRelatedAssembly.class)),
+                        getCavitySealOccurrences().stream().flatMap(StreamUtils.ofClass(HasRelatedAssembly.class)),
+                        getCoPackOccurrences().stream().flatMap(StreamUtils.ofClass(HasRelatedAssembly.class)),
+                        getComponentBoxOccurrences().stream().flatMap(StreamUtils.ofClass(HasRelatedAssembly.class)),
+                        getComponentOccurrences().stream().flatMap(StreamUtils.ofClass(HasRelatedAssembly.class)),
+                        getConnectorOccurrences().stream().flatMap(StreamUtils.ofClass(HasRelatedAssembly.class)),
+                        getFixingOccurrences().stream().flatMap(StreamUtils.ofClass(HasRelatedAssembly.class)),
+                        getSpecialTerminalOccurrences().stream().flatMap(StreamUtils.ofClass(HasRelatedAssembly.class)),
+                        getGeneralWireOccurrences().stream().flatMap(StreamUtils.ofClass(HasRelatedAssembly.class)),
+                        getTerminalOccurrences().stream().flatMap(StreamUtils.ofClass(HasRelatedAssembly.class)),
+                        getWireProtectionOccurrences().stream().flatMap(StreamUtils.ofClass(HasRelatedAssembly.class)))
+                .flatMap(Function.identity())
+                .collect(Collectors.toList());
+    }
```

Comment:
Would create a private helper method to do the `stream.flatMap(...)` part.

Since Java 9, private methods in interfaces are valid now.

```java
private <T> List<HasRelatedAssembly> getRelatedAssemblies(List<T> list)  {
    return list.stream()
            .flatMap(StreamUtils.ofClass(HasRelatedAssembly.class))
            .collect(Collectors.toList());
}
```